integration/yarn/src/main/java/tachyon/yarn/Client.java

```diff
@@ -0,0 +1,388 @@
+            .addArg(mAppMasterMainClass).addArg(mNumWorkers).addArg(mTachyonHome)
+            .addArg(mMasterAddress)
+            .addArg("1>" + ApplicationConstants.LOG_DIR_EXPANSION_VAR + "/stdout")
+            .addArg("2>" + ApplicationConstants.LOG_DIR_EXPANSION_VAR + "/stderr").toString();
+
+    System.out.println("AM command: " + amCommand);
+    mAmContainer.setCommands(Collections.singletonList(amCommand));
+
+    // Setup jar for ApplicationMaster
+    LocalResource appMasterJar = Records.newRecord(LocalResource.class);
+    setupAppMasterJar(appMasterJar);
+    mAmContainer.setLocalResources(Collections.singletonMap("tachyon.jar", appMasterJar));
+
+    // Setup CLASSPATH for ApplicationMaster
+    Map<String, String> appMasterEnv = new HashMap<String, String>();
+    setupAppMasterEnv(appMasterEnv);
+    mAmContainer.setEnvironment(appMasterEnv);
+  }
+
+  private void setupAppMasterJar(LocalResource appMasterJar) throws IOException {
+    Path jarPath = new Path(mHDFSAppMasterJar); // known path to jar file on HDFS
+    FileStatus jarStat = FileSystem.get(mYarnConf).getFileStatus(jarPath);
+    appMasterJar.setResource(ConverterUtils.getYarnUrlFromPath(jarPath));
+    appMasterJar.setSize(jarStat.getLen());
+    appMasterJar.setTimestamp(jarStat.getModificationTime());
+    appMasterJar.setType(LocalResourceType.FILE);
+    appMasterJar.setVisibility(LocalResourceVisibility.PUBLIC);
+  }
+
+  private void setupAppMasterEnv(Map<String, String> appMasterEnv) {
+    for (String c : mYarnConf.getStrings(YarnConfiguration.YARN_APPLICATION_CLASSPATH,
+        YarnConfiguration.DEFAULT_YARN_APPLICATION_CLASSPATH)) {
+      Apps.addToEnvironment(appMasterEnv, ApplicationConstants.Environment.CLASSPATH.name(),
+          c.trim(), ApplicationConstants.CLASS_PATH_SEPARATOR);
+    }
+    Apps.addToEnvironment(appMasterEnv, ApplicationConstants.Environment.CLASSPATH.name(),
+        Environment.PWD.$() + File.separator + "*", ApplicationConstants.CLASS_PATH_SEPARATOR);
+  }
+
+  /**
+   * Sets up the application submission context.
+   */
+  private void setupApplicationSubmissionContext() {
+    // set the application name
+    mAppContext.setApplicationName(mAppName);
+
+    // Set up resource type requirements
+    // For now, both memory and vcores are supported, so we set memory and vcores requirements
+    Resource capability = Resource.newInstance(mAmMemory, mAmVCores);
+    mAppContext.setResource(capability);
+
+    // Set the queue to which this application is to be submitted in the RM
+    mAppContext.setQueue(mAmQueue);
+
+    // Set the AM container spec
+    mAppContext.setAMContainerSpec(mAmContainer);
+
+    // Set the priority for the application master
+    mAppContext.setPriority(Priority.newInstance(mAmPriority));
+  }
+
+  /**
+   * Monitors the submitted application for completion.
+   *
+   * @return true if application completed successfully
+   * @throws YarnException
```

Comment:
added minimal explanation as that is what I could get from YARN api documentation.